niceman/distributions/debian.py

```diff
@@ -543,35 +529,32 @@ def proc_apt_cache_show_cmd(subqueries, cmd_results):
                     p[f] = r[f]
 
     def _get_pkgs_install_date(self, pkg_dicts):
-        # Define a function to process batches of packages and return
-        # a dict from dpkg list filename to creation date
-        def proc_stat_cmd(subqueries, cmd_results):
-            try:
-                out, _ = self._session.execute_command(
-                    ['stat', '-c', '%n: %Y'] + subqueries
-                )
-            except CommandError as exc:  # file not found
+        # Convert package names to dpkg list filenames
+        queries = []
+        for p in pkg_dicts:
+            queries.append(self._pkg_name_to_dpkg_list_file(p["name"]))
+        # Call stat in batches
+        exec_gen = execute_command_batch(self._session,
+                                         ['stat', '-c', '%n: %Y'],
+                                         queries,
+                                         CommandError)
+        # Parse and accumulate stat results in a dict
+        results = {}
+        for (out, _, exc) in exec_gen:
+            if exc:
                 stderr = utils.to_unicode(exc.stderr, "utf-8")
                 if 'No such file or directory' in stderr:
                     out = exc.stdout  # One file not found, so continue
                 else:
-                    raise  # some other fault -- handle it above
+                    raise exc  # some other fault -- handle it above
+            out = utils.to_unicode(out, "utf-8")
```

Comment:
more I see of the same lines, more I am confident that probably should be a part of the `execute_command_batch` command

Reply:
Yup (this forced me to update my test_execute_command_batch because my mock returned an int....  Shame on me for making a mock that didn't match the function signature :/